test/extended/cli/template.go

```diff
@@ -16,10 +18,299 @@ var _ = g.Describe("[sig-cli] templates", func() {
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		outputTemplateFile, err := oc.Run("process").Args("-f", guestbookTemplatePath, "-l", "app=guestbook").OutputToFile("guestbooktemplate.yaml")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("create").Args("-f", outputTemplateFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err = oc.Run("status").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("frontend-service"))
+		out, err = oc.Run("process").Args("-f", guestbookTemplatePath, "--local", "-l", "app=guestbook", "-o", "yaml").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("app: guestbook"))
+		out, err = oc.Run("process").Args("-f", guestbookTemplatePath, "--local", "-l", "app=guestbook", "-o", "yaml", "--server", "0.0.0.0:1").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("app: guestbook"))
+		err = oc.Run("delete").Args("-f", outputTemplateFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("individually specified parameter values are honored")
+		out, err = oc.Run("process").Args("-f", guestbookTemplatePath, "-p", "ADMIN_USERNAME=myuser", "-p", "ADMIN_PASSWORD=mypassword").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("myuser"))
+		o.Expect(out).To(o.ContainSubstring("mypassword"))
+		out, err = oc.Run("process").Args("ADMIN_USERNAME=myuser", "ADMIN_PASSWORD=mypassword", "-f", guestbookTemplatePath).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("myuser"))
+		o.Expect(out).To(o.ContainSubstring("mypassword"))
+		err = oc.Run("create").Args("-f", appTemplateStiPath).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "MYSQL_USER=myself", "MYSQL_PASSWORD=my,1%pa=s").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("myself"))
+		out, err = oc.Run("process").Args("MYSQL_USER=myself", "MYSQL_PASSWORD=my,1%pa=s", "ruby-helloworld-sample").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("my,1%pa=s"))
+		out, err = oc.Run("process").Args("ruby-helloworld-sample", "-p", "MYSQL_USER=myself", "-p", "MYSQL_PASSWORD=my,1%pa=s").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("myself"))
+		out, err = oc.Run("process").Args("-p", "MYSQL_USER=myself", "-p", "MYSQL_PASSWORD=my,1%pa=s", "ruby-helloworld-sample").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("my,1%pa=s"))
+		out, err = oc.Run("process").Args("-f", guestbookTemplatePath, fmt.Sprintf("--param-file=%s", guestbookTemplateEnvPath)).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("root"))
+		o.Expect(out).To(o.ContainSubstring("adminpass"))
+		o.Expect(out).To(o.ContainSubstring("redispass"))
```

Comment:
Missing the same but reading the same input from stdin using cat guestbookTemplatePath. Still worth checking as well. I.e.
```
os::cmd::expect_success_and_text "cat '${guestbook_params}' | oc process -f '${guestbook_template}' --param-file=-" '"root"'
os::cmd::expect_success_and_text "cat '${guestbook_params}' | oc process -f '${guestbook_template}' --param-file=-" '"adminpass"'
os::cmd::expect_success_and_text "cat '${guestbook_params}' | oc process -f '${guestbook_template}' --param-file=-" '"redispass"'
```